src/shims/unix/fd.rs

```diff
@@ -310,6 +382,45 @@ impl FdTable {
 
 impl<'tcx> EvalContextExt<'tcx> for crate::MiriInterpCx<'tcx> {}
 pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
+    /// For a specific unique file descriptor id, get its ready events and update the corresponding ready lists
+    fn check_and_update_readiness(
+        &self,
+        id: FdId,
+        get_ready_events: impl FnOnce() -> InterpResult<'tcx, EpollReadyEvents>,
+    ) -> InterpResult<'tcx, ()> {
+        let this = self.eval_context_ref();
+        // Get a list of EpollEventInterest that is associated to a specific file description.
+        if let Some(epoll_interests) = this.machine.epoll_interests.get_epoll_interest(id) {
+            // Retrieve the readiness events of the file description.
+            let epoll_ready_events = get_ready_events()?;
+
+            // Get the bitmask of ready events.
+            let epollin = this.eval_libc_u32("EPOLLIN");
+            let epollout = this.eval_libc_u32("EPOLLOUT");
+            let epollrdhup = this.eval_libc_u32("EPOLLRDHUP");
+            let ready_events = epoll_ready_events.get_event_bitmask(epollin, epollout, epollrdhup);
```

Comment:
could pass the `ecx` into `get_event_bitmask` and call the `eval_libc_u32` methods inside

Reply:
Good idea!